### PR TITLE
fix: months not updating when changing options

### LINKED ISSUE #1069: [Calendar & related] Updating options doesn't update values
### Describe the bug

When you try to update some value of some option that would change the calendar (specifically `numberOfMonths`, `fixedWeeks`, `weekStartsOn`) the calendar is not updated until the month or the date is changed.

I've already investigated this and it all boils down to 

https://github.com/melt-ui/melt-ui/blob/746ee7c65dea907f64af3314d51d7096fa0c0b7f/src/lib/builders/calendar/create.ts#L524-L530

the effect is responsible to recalculate the months but it bails out if the current placeholder is already in the visibleMonths. Removing this solves the issue but i suspect will do a bit more work in general. I think this check should be in place only if the placeholder is the only thing that changed.

I can also work on this and open a PR, just wanted a bit more insight on this if (wink wink @huntabyte )

### Reproduction

To test the fixed weeks not working obviously set to a month were fixed would work

https://www.sveltelab.dev/44fmd3k81k63jen

### Logs

_No response_

### System Info

```bash
  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.3 - /usr/local/bin/pnpm
  npmPackages:
    @melt-ui/svelte: latest => 0.75.3 
    @sveltejs/kit: ^2.0.0 => 2.5.2 
    svelte: ^4.1.2 => 4.2.12 
```

### Severity

blocking an upgrade

### PR DESCRIPTION
**Pull request by paoloricciuti**
Closes #1069

### CODE REVIEW

**Comment by huntabyte** [comment]
Also closes #725

File: src/lib/builders/calendar/create.ts [hunk 1]
```diff
@@ -514,20 +514,48 @@ export function createCalendar<
 	 * Updates the displayed months based on changes in the placeholder value,
 	 * which determines the months to show in the calendar.
 	 */
-	effect(
-		[placeholder, weekStartsOn, locale, fixedWeeks, numberOfMonths],
-		([$placeholder, $weekStartsOn, $locale, $fixedWeeks, $numberOfMonths]) => {
-			if (!isBrowser || !$placeholder) return;
+	effect([placeholder, weekStartsOn, locale, fixedWeeks, numberOfMonths], ([$placeholder]) => {
```
Comment (huntabyte) [inline src/lib/builders/calendar/create.ts:517]:
```suggestion
	effect([placeholder], ([$placeholder]) => {
```

Comment (huntabyte) [reply src/lib/builders/calendar/create.ts:517]: We need to remove these so this only fires when the `placeholder` changes and we aren't doubling with the following effect.

**Review by huntabyte — CHANGES_REQUESTED** [review]

**Comment by paoloricciuti** [comment]
> Also closes #725

Are you sure about that? It seems that is asking of dynamically set the first week of the day based on the locale...is that automatic?

File: src/lib/builders/calendar/create.ts (hunk 1 above)
Comment (paoloricciuti) [reply src/lib/builders/calendar/create.ts:517]: Duh I'm stupid lol...I removed initially, then CMD+Z to copy and forgot to re-remove it lol

**Comment by huntabyte** [comment]
> > Also closes #725
> 
> Are you sure about that? It seems that is asking of dynamically set the first week of the day based on the locale...is that automatic?

We can add a test to confirm but I believe it's because the month wasn't being recreated (which handles that automatically if not specified)

**Comment by paoloricciuti** [comment]
> > > Also closes #725
> > 
> > 
> > Are you sure about that? It seems that is asking of dynamically set the first week of the day based on the locale...is that automatic?
> 
> We can add a test to confirm but I believe it's because the month wasn't being recreated (which handles that automatically if not specified)

Do you know of a locale that has a different starting day?

**Comment by paoloricciuti** [comment]
> > > Also closes #725
> > 
> > 
> > Are you sure about that? It seems that is asking of dynamically set the first week of the day based on the locale...is that automatic?
> 
> We can add a test to confirm but I believe it's because the month wasn't being recreated (which handles that automatically if not specified)

Ok checked and it's true...adding a test for that too

**Review by huntabyte — APPROVED** [review]
You're incredible! Can we make the same changes to the Range Calendar? If you don't have time I can take care of it!

**Comment by paoloricciuti** [comment]
> You're incredible! Can we make the same changes to the Range Calendar? If you don't have time I can take care of it!

I should be able to do it

**Pull request merged**